data/tokenizer.py

```diff
@@ -184,14 +174,10 @@ def quantize(self, start: list[float]) -> list[int]:
         return dstart_bins
 
     def tokenize(self, record: dict) -> list[str]:
-        # TODO I don't love the idea of adding tokens durint *tokenize* call
-        # If we want to pretend that our midi sequences have start and finish
-        # we should take care of that before we get here :alarm:
         dstart_bins = self.quantize(record["start"])
 
         # get tokens from quantized data
-        tokens = [str(dstart_bin) for dstart_bin in dstart_bins]
-        tokens = ["<s>"] + tokens + ["</s>"]
+        tokens = [""] + [str(dstart_bin) for dstart_bin in dstart_bins]
```

Comment:
```suggestion
        tokens = [str(dstart_bin) for dstart_bin in dstart_bins]
```

Removing my comment about not adding tokens in this function does not address the issue 😅 

I think special tokens should be added in a step separated from tokenization of `dstart/velocity` bins, let me know if you see good reasons not to do this, because maybe I'm missing something.

Also, this first token shouldn't be an empty string, but rather a "<CLS>" token. This will make it consistent with other transformer-oriented research that we're doing.

Reply:
Sorry, I deleted it when I was removing all the special tokens and thought it is resolved :sweat_smile: 

I am not convinced as to the necessity of adding a \<CLS> token before tokenization. We may add it after, but the existance of this token derives directly form it's use inside the transformer and would not make any sense in an untokenized musiacal sequence.

As for first token - I might have misunderstood your previeous comment - github hid the cls token :0.
> I see, let's rename it to "" for now - this will be consistent with BERT approach. We can later try to use it to do composer classification or other downstream tasks